### ISSUE TITLE
Vennam’s noisy-gradient explanation is a hypothesis, not a demonstrated cause of thinking-token underperformance

### ISSUE THREAD

**Issue by microprediction** [microprediction/latentreasoning#128]
## Reviewed source

- Default-branch commit: `88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b`
- Affected text: [`bibliography.html` lines 640–644](https://github.com/microprediction/latentreasoning/blob/88d1db6b1c014f641f67ec4b84ec1f51e8f0c43b/bibliography.html#L640-L644)
- Primary source: Vennam et al., [*Rethinking Thinking Tokens: Understanding Why They Underperform in Practice*, arXiv:2411.11371v1](https://arxiv.org/html/2411.11371)

The bibliography currently says:

> “Naive thinking tokens consistently underperform real chain of thought: a single repeated embedding produces noisy, inconsistent gradients.”

The first clause is a fair summary of the paper’s small comparison. The causal clause states the paper’s hypothesis as an established mechanism.

## What the experiments establish

Table 1 compares four training configurations on:

- a GPT-2-style model trained on synthetic 2-, 3-, and incompletely run 4-digit multiplication; and
- Llama-3.2-1B fine-tuned on GSM8K and OpenBookQA.

Within these runs, one thinking-token configuration is below supervised CoT on every reported task. It improves the multiplication baseline, hurts the GSM8K baseline, and ties the OpenBookQA baseline. The paper reports no repeated-seed uncertainty or confidence intervals.

That supports a scoped empirical result: **this implementation did not match supervised CoT in these settings**.

## The noisy-gradient mechanism is not causally identified

The paper labels its explanation a hypothesis in the abstract and §3. Its gradient evidence consists principally of:

1. the one-token embedding moving little from initialization;
2. its cumulative gradient vector being small;
3. two distinct thinking-token embeddings showing larger cumulative gradients and more movement.

Those observations do not establish that gradient “noise” caused the accuracy gap.

If a shared embedding `e` appears at positions `p`, parameter sharing gives:

```text
∇ₑ L = Σₚ ∂L/∂eₚ
```

This is ordinary tied-parameter differentiation; vocabulary embeddings and transformer weights are also reused across many contexts. Cancellation may reflect conflicting tasks, weak or non-identifiable signal, convergence near a stationary point, optimizer dynamics, or actual stochastic noise. A small cumulative vector alone cannot distinguish these explanations. The paper assumes that noise is zero-mean (§4.2) but does not estimate a gradient signal-to-noise ratio, within-/between-example covariance, or a causal mediation from gradient statistics to task performance.

The paper’s Eq. 4 also writes a sum over `k` indexed thinking-token embeddings while the prose attributes the problem to repeated uses of one embedding. That notation does not derive why one shared embedding must be noisy; it merely restates that gradients add under parameter sharing.

Most importantly, the two-embedding experiment reports clearer gradients and greater embedding movement, but **does not report downstream task accuracy for the two-embedding model**. It therefore never shows that the proposed remedy closes any performance gap. The TT+CoT row is a different intervention and does not isolate the single- versus two-embedding mechanism.

## Consequence

The bibliography currently turns an underdetermined diagnostic into the cause of thinking-token underperformance. This matters because later pause/filler-token work uses multiple training schedules, frozen or contentless embeddings, position encodings, and larger models; Vennam et al.’s result does not establish a general failure mechanism for repeated embeddings.

## Concrete correction

For example:

> In one GPT-2 multiplication setup and Llama-3.2-1B fine-tuning on GSM8K/OpenBookQA, a single thinking-token configuration underperforms supervised CoT. The authors hypothesize destructive or noisy gradient aggregation because its shared embedding moves little and has a small cumulative gradient; two embeddings move more, but no two-embedding task accuracy, seed uncertainty, or gradient-SNR analysis is reported, so the mechanism is not causally established.

I searched open and closed issues for the title, arXiv identifier, Vennam, thinking tokens, and noisy gradients and found no existing issue covering this distinction.

**Comment by microprediction** [comment]
Confirmed and applied in 5f7ecd7. The abstract and §3 frame noisy gradients as a hypothesis. The evidence in §4.2/§5.1 is how far the embedding moves and the cumulative gradient, and the two-embedding run reports only those, with no task accuracy. Table 1 covers GPT-2 multiplication and Llama 3.2 1B on GSM8K and OpenBookQA. The bibliography, families page, map and timeline now present the mechanism as the authors' untested hypothesis.
